cluster.go

```diff
@@ -108,8 +109,15 @@ func (c *ClusterClient) slotMasterAddr(slot int) string {
 // randomClient returns a Client for the first live node.
 func (c *ClusterClient) randomClient() (client *Client, err error) {
 	for i := 0; i < 10; i++ {
-		n := rand.Intn(len(c.addrs))
-		client, err = c.getClient(c.addrs[n])
+		var addr string
+		if len(c.masters) > 0 {
```

Comment:
I had to use masters here, becuase it looks like slaves don't propagate pub/sub messages to masters.